cibuildwheel/macos.py

```diff
@@ -409,7 +409,9 @@ def build(options: Options, tmp_path: Path) -> None:
                                 config.version
                             )
                         )
-                        build_env["PIP_CONSTRAINT"] = constraint_path.as_uri()
+                        build_env["PIP_CONSTRAINT"] = (
+                            constraint_path.as_uri() + " " + build_env.get("PIP_CONSTRAINT", "")
+                        )
```

Comment:
```suggestion
                        constraint = build_env.get("PIP_CONSTRAINT", "")
                        build_env["PIP_CONSTRAINT"] = f"{constraint_path.as_uri()} {constraint}"
```

Reply:
Actually, I don't like that the above adds a space to the end if PIP_CONSTRAINT is not set. We could add `.strip()`, or just build it correctly:

```suggestion
                        constraints = [constraint_path.as_uri(), build_env.get("PIP_CONSTRAINT", None)]
                        build_env["PIP_CONSTRAINT"] = " ".join(filter(constraints, None))
```